### PR TITLE
[BugFix] Decode DreamerV3 rewards and values end to end

### PR DESCRIPTION
**Pull request by vmoens**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.17.0) (oldest at bottom):
* #4095
* #4075
* #4074
* #4073
* #4072
* #4068
* #4067
* __->__ #4066
* #4065

Summary:
- add configurable reward_logits and value_logits NestedKeys while reserving reward and value keys for decoded scalars
- preserve legacy logits-under-scalar-key inputs with v0.16 deprecation warnings
- expose categorical critic logits and decoded values to lambda returns
- share the trained prior and categorical reward head with imagination
- configure the maintained DreamerV3 example for 255-bin reward and value cross entropy

Rationale:
Training consumes categorical logits, whereas environment rewards, critic baselines, and TD-lambda returns require decoded scalars. Separating these representations prevents logits from leaking into return computation, and sharing the trained heads ensures imagination reflects the optimized world model.

Test plan:
- uv run pytest -q test/objectives/test_dreamer_v3.py
- run the maintained DreamerV3 SOTA smoke with 11-bin CI overrides
- flake8 --config=setup.cfg on the changed Python files
- ufmt check on the changed Python files
- git diff --check

### CODE REVIEW

**Comment by vmoens** [comment]
**Update after the stack-level ghstack rebase:** The full eight-PR Dreamer stack was checked out from #4075, rebased onto current `origin/main`, linted with `pre-commit run --all-files`, and submitted successfully as `vmoens`. This PR is now at `ba23945a1465`. The core Dreamer suites pass (267 tests) and the two doctest modules pass (15 tests); the DMC-only smoke still requires `dm_control`, which is not installed locally. Any earlier conflict/head-status note below is superseded; the substantive code-review observations remain applicable.

Reviewed published head `6df1724eea04`, including the full layer diff, the latest commit, current checks, and the complete PR conversation (there are no human review threads yet). I also attempted the requested rebase after fetching `origin/main`, but it stopped on a content conflict in `sota-implementations/dreamer_v3/dreamer_v3.py`; per the rebase workflow, I left that rebase in progress and did not push a partial result.

Within the published patch, the separation of categorical logits from decoded reward/value keys is coherent: the loss reads the configurable logits keys, the legacy fallback has an explicit v0.16 deprecation, nested-key coverage is present, and imagination shares the trained prior/reward head instead of independent parameters. I did not find an additional code defect in this layer. The four new focused tests (decoded critic value, legacy warning, nested logits keys, and shared imagination parameters) pass, and `git diff --check` is clean.

Residual risk: this head is still based on the pre-rebase stack, so it has not been validated against current `origin/main`. The existing bot-reported lint failure is an infrastructure/container failure rather than a source diagnostic. Please resolve the DreamerV3 script conflict and rerun the focused suite plus the SOTA smoke on the rebased result.

**Comment by vmoens** [comment]
Addressed the review pass and resubmitted this commit after rebasing the stack onto current main. No layer-specific code change was needed; the categorical logits remain separated from decoded reward/value scalars and the shared imagination heads are preserved. The full all-files lint suite passes at this PR boundary.
